packages/rest-hooks/src/react-integration/hooks/useRetrieve.ts

```diff
@@ -1,15 +1,15 @@
-import { ReadShape, Schema } from 'rest-hooks/resource';
+import { ReadShape, ParamsFromShape } from 'rest-hooks/resource';
 
 import { useMemo } from 'react';
 
 import useFetcher from './useFetcher';
 import useExpiresAt from './useExpiresAt';
 
 /** Request a resource if it is not in cache. */
-export default function useRetrieve<
-  Params extends Readonly<object>,
-  S extends Schema
->(fetchShape: ReadShape<S, Params>, params: Params | null) {
+export default function useRetrieve<Shape extends ReadShape<any, any>>(
```

Comment:
parameter of Shape rather than a part gives us more information. this is used in useFetcher() which infers its return type